lib/web_ui/lib/src/engine/canvaskit/renderer.dart

```diff
@@ -409,8 +414,53 @@ class CanvasKitRenderer implements Renderer {
   ui.ParagraphBuilder createParagraphBuilder(ui.ParagraphStyle style) =>
     CkParagraphBuilder(style);
 
+  ({
+    ui.Scene scene,
+    ui.FlutterView view,
+    Completer<void> completer
+  })? _currentRender;
+  ({
+    ui.Scene scene,
+    ui.FlutterView view,
+    Completer<void> completer
+  })? _nextRender;
+
+  // TODO(harryterkelsen): Merge this logic with the async logic in
+  // [EngineScene], https://github.com/flutter/flutter/issues/142072.
   @override
   Future<void> renderScene(ui.Scene scene, ui.FlutterView view) async {
+    if (_currentRender != null) {
+      // If a scene is already queued up, drop it and queue this one up instead
+      // so that the scene view always displays the most recently requested scene.
+      _nextRender?.completer.complete();
```

Comment:
Can this break multi-view? The framework may render multiple `Scene` objects into different views, all in the same frame (e.g. every view contains an animation). That would queue up multiple renders in the same frame.

If this is indeed an issue, we could store the `currentRender` and `nextRender` on the `FlutterView` object instead (or on the `ViewRasterizer`). This way, every view has its own queue that doesn't interfere with other views.

Reply:
Good catch. I'll make the queues per-view and add a test.

Reply:
Done.